site/_js/extension-samples.js

```diff
@@ -0,0 +1,295 @@
+          pillElement.classList.add(
+            'more-api-pill',
+            'tag-pill',
+            'surface',
+            'hairline',
+            'color-secondary-text',
+            'type--label',
+            'weight-regular',
+            'rounded-lg'
+          );
+          pillElement.setAttribute('href', apiDocUrl);
+          pillElement.setAttribute('target', '_blank');
+          pillElement.setAttribute('rel', 'noopener');
+          pillElement.textContent = `${api.namespace}.${api.propertyName}`;
+          fragment.appendChild(pillElement);
+        }
+        showMoreButton.parentNode?.insertBefore(fragment, showMoreButton);
+        showMoreButton.textContent = showLessText;
+      } else {
+        const moreApiPills = sampleElement.querySelectorAll('.more-api-pill');
+        moreApiPills.forEach(pill => pill.remove());
+        showMoreButton.textContent = showMoreText;
+      }
+    });
+  });
+
+  selectFields.forEach(ele => {
+    ele.addEventListener('change', e => {
+      const t = e.target;
+
+      if (!(t instanceof EnhancedSelect)) {
+        return;
+      }
+
+      activeFilters[t.name] = t.value;
+
+      updateUi();
+      updateTagPills();
+    });
+  });
+
+  searchInput?.addEventListener('input', e => {
+    if (!(e instanceof InputEvent)) return;
+
+    if (e.isComposing) return;
+    searchKeyword = /** @type { HTMLInputElement } */ (e.target).value.trim();
+    updateUi();
+  });
+
+  document
+    .querySelector('#search-extension-samples-close')
+    ?.addEventListener('click', () => {
+      /** @type { HTMLInputElement } */ (searchInput).value = '';
+      searchKeyword = '';
+      updateUi();
+    });
+
+  activeFiltersList?.addEventListener('removed-pill', e => {
+    if (!(e instanceof CustomEvent)) return;
+
+    const index = activeFilters[e.detail.key].indexOf(e.detail.value);
+
+    activeFilters[e.detail.key].splice(index, 1);
+
+    updateUi();
+
+    selectFields.forEach(field => {
+      /** @type { EnhancedSelect } */ (field).setValue(
+        activeFilters[field.getAttribute('name')] || []
+      );
+    });
+
+    document
+      .querySelectorAll('#mobile-filters input[type="checkbox"]:checked')
+      .forEach(checkbox => {
+        const target = activeFilters?.[checkbox.getAttribute('name')];
+
+        /** @type { HTMLInputElement } */ (checkbox).checked =
+          target && target.includes(checkbox.getAttribute('value'));
+      });
+  });
+}
+
+function initItemData() {
+  const dataWrapper = document.getElementById('extension-samples-data');
+  if (!dataWrapper) {
+    throw new Error('Missing data wrapper');
+  }
+  const data = JSON.parse(dataWrapper.innerText);
+  const items = document.querySelectorAll('.sample-item');
+  items.forEach(item => {
+    const name = item.getAttribute('data-sample-name');
+    if (!name) {
+      throw new Error('Missing sample name');
+    }
+    const sample = data.find(i => i.name === name);
```

Comment:
Nit, but if `data` was an object keyed by the names, this could be a simple lookup rather than a find which is essentially a nested loop.